CMakeLists.txt

```diff
@@ -0,0 +1,263 @@
+    set(HAVE_WXWIDGETS OFF)
+endif()
+
+# Find optional dependencies
+if(ENABLE_PYTHON)
+    find_package(Python3 COMPONENTS Development)
+    if(Python3_FOUND)
+        message(STATUS "Found Python: ${Python3_VERSION}")
+        add_definitions(-DHAVE_PYTHON)
+    else()
+        message(WARNING "Python development headers not found. Python support will be disabled.")
+        set(ENABLE_PYTHON OFF)
+    endif()
+endif()
+
+if(ENABLE_SSL)
+    find_package(OpenSSL)
+    if(OpenSSL_FOUND)
+        message(STATUS "Found OpenSSL: ${OPENSSL_VERSION}")
+        add_definitions(-DHAVE_SSL)
+    else()
+        message(WARNING "OpenSSL not found. SSL/TLS support will be disabled.")
+        set(ENABLE_SSL OFF)
+    endif()
+endif()
+
+# Platform-specific libraries
+if(WIN32)
+    set(PLATFORM_LIBS winmm comctl32 rpcrt4 ws2_32)
+elseif(UNIX)
+    set(PLATFORM_LIBS dl pthread)
+    if(NOT APPLE)
+        list(APPEND PLATFORM_LIBS rt)
+    endif()
+endif()
+
+# Set global include directories
+include_directories(
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+    ${CMAKE_CURRENT_BINARY_DIR}/include
+)
+
+# Configure version header
+configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/include/Mversion.h.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/include/Mversion.h"
+    @ONLY
+)
+
+# Add subdirectories for libraries
+add_subdirectory(lib/imap)
+add_subdirectory(lib/compface)
+add_subdirectory(lib/dspam)
+add_subdirectory(src/wx/vcard)
+
+# Add main source directory (only if wxWidgets is available)
+if(HAVE_WXWIDGETS)
+    add_subdirectory(src)
+else()
+    message(STATUS "Skipping main application build (wxWidgets not found)")
+endif()
+
+# Add tests if enabled
+if(BUILD_TESTS)
+    enable_testing()
+    add_subdirectory(tests)
+endif()
+
+# Install configuration
+include(GNUInstallDirs)
+
+# Generate and install export configuration
+install(EXPORT MahoganyTargets
+    FILE MahoganyTargets.cmake
+    NAMESPACE Mahogany::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Mahogany
+)
+
+# Create config file
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/cmake/MahoganyConfig.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/MahoganyConfig.cmake"
+    INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Mahogany
+)
+
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/MahoganyConfigVersion.cmake"
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMajorVersion
+)
+
+install(FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/MahoganyConfig.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/MahoganyConfigVersion.cmake"
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Mahogany
+)
+
+# Install headers
+install(DIRECTORY include/
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mahogany
+    FILES_MATCHING PATTERN "*.h"
+)
+
+# Install configured version header
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/include/Mversion.h"
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mahogany
+)
```

Comment:
We don't need to install all this, Mahogany is an application and not a library and won't be used by any other package.